AmplifyPlugins/Analytics/AWSPinpointAnalyticsPluginIntegrationTests/AWSPinpointAnalyticsPluginIntegrationTests.swift

```diff
@@ -72,11 +72,39 @@ class AWSPinpointAnalyticsPluginIntergrationTests: XCTestCase {
         wait(for: [identifyUserEvent], timeout: TestCommonConstants.networkTimeout)
 
         //TODO: Fix the test below
-        //        // Remove userId from the current endpoint
-        //        let targetingClient = escapeHatch().targetingClient
-        //        let currentProfile = targetingClient.currentEndpointProfile()
-        //        currentProfile.user?.userId = ""
-        //        targetingClient.update(currentProfile)
+        // Remove userId from the current endpoint
+        // let targetingClient = escapeHatch().targetingClient
+        // let currentProfile = targetingClient.currentEndpointProfile()
+        // currentProfile.user?.userId = ""
+        // targetingClient.update(currentProfile)
+    }
+
+    /// Run this test when the number of endpoints for the userId exceeds the limit.
+    /// The profile should have permissions to run the "mobiletargeting:DeleteUserEndpoints" action.
+    func skip_testDeleteEndpointsForUser() throws {
+        let userId = "userId"
+        let escapeHatch = escapeHatch()
+        let applicationId = escapeHatch.configuration.appId
+        guard let targetingConfiguration = escapeHatch.configuration.targetingServiceConfiguration else {
+            XCTFail("Targeting configuration is not defined.")
+            return
+        }
+
+        let deleteEndpointsRequest = AWSPinpointTargetingDeleteUserEndpointsRequest()!
+        deleteEndpointsRequest.userId = userId
+        deleteEndpointsRequest.applicationId = applicationId
+
+        let deleteExpectation = expectation(description: "Delete endpoints")
+        let lowLevelClient = lowLevelClient(from: targetingConfiguration)
+        lowLevelClient.deleteUserEndpoints(deleteEndpointsRequest) { response, error in
+            guard error == nil else {
+                XCTFail("Unexpected error when attempting to delete endpoints")
+                deleteExpectation.fulfill()
+                return
+            }
+            deleteExpectation.fulfill()
+        }
+        wait(for: [deleteExpectation], timeout: 1)
```

Comment:
Recommend we confirm with @ruisebas or @phantumcode to confirm this should be brought into dev-preview

Reply:
per @ruisebas, this needs to be removed.